carvis-web/src/server/routes/redisRoutes.js

```diff
@@ -0,0 +1,95 @@
+/* ======== REDIS SCHEMA =====
+places:
+---------
+hash name: place (string)
+keys:
+---
+userId:lat : int
+userId:lng : int
+userId:routable : string
+
+-- otherwise do getOwnPropertyNames to check another userId:data
+// NOTE: or is there a better way?
+
+===============================
+userdata:
+---------
+hash name: userId (alexa userId)
+keys:
+---
+lyftToken: string
+lyftPaymentInfo: string
+uberToken: string
+originlat: int
+originlng: int
+originRoutable: string
+phoneNumber: int
+...
+
+===============================
+
+*/
+
+
+var bluebird = require("bluebird");
+bluebird.promisifyAll(require('redis')); // promisified versions have 'Async' postfix - ie. client.getAsync() and client.hgetallAsync()
+var client = redis.createClient();
+
+// this sets the Redis server as an LRU cache with 500MB space.
+client.config("SET", "maxmemory", "500mb"); // should this be bigger ?
```

Comment:
yeah space is expensive - 500MB is more than enough tbh.

so flow for when we have userId (alexa skill) is straightforward.

flow on web client (not main case) -- we could store web client user on cache as well by a different property (not many fields are stored) for users who frequently go via the web client, after an initial DB lookup for that user. Since Redis is an LRU we automatically deprecate not frequently used data, so I don't think we have to worry too much about this.